src/batHashtbl.mlv

```diff
@@ -183,16 +183,38 @@ let length h = (h_conv h).size
 
 let is_empty h = length h = 0
 
+exception Hashtbl_key_not_found
 let modify_opt key f h =
-  match find_option h key with
-  | Some _ as v ->
-     (match f v with
-      | Some v' -> replace h key v'
-      | None -> remove h key)
-  | None ->
-     (match f None with
-      | Some v -> add h key v
-      | None -> ())
+  let hc = h_conv h in
+
+  let rec loop = function
+    | Empty ->
+      (* Inserting an element might require a resize of the hash table.
+         We rely on Hashtbl.add function to grow the hashtbl if needed
+         instead of duplicating logic from the OCaml standard library. *)
```

Comment:
Sorry for the nitpicking, but this comment should be above the `| Empty` line, not under it (as placed it seems to comment on the `raise Not_found ...` line below, or in any case be specific to the `Empty` case. If you are uneasy with separating the head `function` and its first case with a comment, feel free to have the comment first, and `function` second on its own line (the `|` does not need to be indented wrt. `function`).